private/bufpkg/bufconfig/file.go

```diff
@@ -205,13 +206,15 @@ func newDecodeError(fileName string, err error) error {
 		fileName = "config file"
 	}
 	// We intercept PathErrors in buffetch to deal with fixing of paths.
-	return &fs.PathError{Op: "decode", Path: fileName, Err: err}
+	// We return a cleaned, unnormalized path in the error for clarity with user's filesystem.
+	return &fs.PathError{Op: "decode", Path: filepath.Clean(normalpath.Unnormalize(fileName)), Err: err}
```

Comment:
`fileName` isn't necessarily a path I believe

Reply:
In the case where it isn't a path (e.g. `fileName = "config file"` or some other case), `normalpath.Unnormalize` and `filepath.Clean` would be no-ops.

Reply:
We need to make a decision as to whether paths should be unnormalized in `fs.PathErrors` - I've generally taken the position that they should not be in the rest of the codebase, I'm worried that we're special-casing here, any thoughts?

Reply:
I'm fine with either -- the unnormalized error was mostly for UX reasons (matching file paths, and we did show unnormalised file paths in the past, etc.) but as noted, they're not always a file path, and consistency is kind of the key. So I'm totally okay with keeping them as normalised paths if this is the pattern throughout the new libraries, in which case, I can adjust the tests to reflect that also.